src/index.ts

```diff
@@ -83,77 +75,114 @@ export function latexRequest<T>(
-      const dirName = PathExt.dirname(widget.context.path);
-      const baseName = PathExt.basename(widget.context.path, '.tex');
-      const pdfFilePath = PathExt.join(dirName, baseName + '.pdf');
-
-      let pdfContext: DocumentRegistry.IContext<DocumentRegistry.IModel>;
-      let errorPanel: ErrorPanel | null = null;
-
-      // Hook up an event listener for when the '.tex' file is saved.
-      widget.context.fileChanged.connect((sender, args) => {
-        latexRequest(widget.context.path, serverSettings).then(() => {
-          // Read the pdf file contents from disk.
-          if (pdfContext) {
-            pdfContext.revert();
-          } else {
-            const pdfWidget = manager.openOrReveal(pdfFilePath);
-            pdfContext = manager.contextForWidget(pdfWidget);
-          }
-          if (errorPanel) {
-            errorPanel.close();
-            errorPanel.dispose();
-            errorPanel = null;
-          }
-        }).catch((err) => {
-          // If there was an error, read the log
-          // file from disk and show it.
-          if (!errorPanel) {
-            errorPanel = new ErrorPanel();
-            errorPanel.id = `latex-error-${++Private.id}`;
-            errorPanel.title.label = 'LaTeX Error';
-            errorPanel.title.closable = true;
-            app.shell.addToMainArea(errorPanel, { ref: widget.id });
-          }
-          errorPanel.text = err.xhr.response;
-        });
-      });
 
-      // Run an initial latexRequest so that the appropriate files exist,
-      // then open them.
-      latexRequest(widget.context.path, serverSettings).then(() => {
-        // Open the pdf and get a handle on its document context.
-        const pdfWidget = manager.openOrReveal(pdfFilePath);
-        pdfContext = manager.contextForWidget(pdfWidget);
+  // Given an fileEditor widget that hosts
+  // a .tex document, open a LaTeX preview for it.
+  const openPreview = (widget: DocumentRegistry.IReadyWidget) => {
+    // If we can't find the document context, bail.
+    let texContext = manager.contextForWidget(widget);
+    if (!texContext) {
+      return;
+    }
+    // If there is already an active preview for this context, bail.
+    if (Private.previews.has(texContext.path)) {
+      return;
+    }
+
+    // build pdfFilePath so that we know what to watch for
+    const dirName = PathExt.dirname(texContext.path);
+    const baseName = PathExt.basename(texContext.path, '.tex');
+    const pdfFilePath = PathExt.join(dirName, baseName + '.pdf');
+
+    let pdfContext: DocumentRegistry.IContext<DocumentRegistry.IModel>;
+    let errorPanel: ErrorPanel | null = null;
+
+    // Hook up an event listener for when the '.tex' file is saved.
+    texContext.fileChanged.connect((sender, args) => {
+      latexRequest(texContext.path, serverSettings).then(() => {
+        // Read the pdf file contents from disk.
+        if (pdfContext) {
+          pdfContext.revert();
```

Comment:
what does `revert()` do here?

Reply:
This is not a change. revert loads the contents from disk.